pkg/controller/pravegacluster/pravegacluster_controller.go

```diff
@@ -329,6 +351,29 @@ func (r *ReconcilePravegaCluster) deployController(p *pravegav1beta1.PravegaClus
 	if err != nil && !errors.IsAlreadyExists(err) {
 		return err
 	}
+	if !eq {
```

Comment:
Same comment adding too many ifs... complicates the flow of code.. I think its time to refactor the reconcile loop...

Reply:
Done.